netutils/dhcpc/dhcpc.c

```diff
@@ -445,6 +448,47 @@ static uint8_t dhcpc_parsemsg(FAR struct dhcpc_state_s *pdhcpc, int buflen,
   return 0;
 }
 
+/****************************************************************************
+ * Name: dhcpc_run
+ ****************************************************************************/
+
+static void *dhcpc_run(void *args)
+{
+  FAR struct dhcpc_state_s *pdhcpc = (FAR struct dhcpc_state_s *)args;
+  struct dhcpc_state result;
+  int ret;
+
+  while (1)
+    {
+      ret = dhcpc_request(pdhcpc, &result);
+      if (ret == OK)
+        {
+          pdhcpc->callback(&result);
+        }
+      else
+        {
+          pdhcpc->callback(NULL);
+          nerr("dhcpc_request error\n");
+        }
+
+      if (pdhcpc->cancel)
+        {
+          return NULL;
+        }
+
+      while (result.lease_time)
+        {
+          result.lease_time = sleep(result.lease_time);
```

Comment:
Note: This should only sleep half of the lease_time before renewing.

Reply:
@coffee809721232

Reply:
> Note: This should only sleep half of the lease_time before renewing.

Yes, it should be half of the time to re-request.
This bug will be fixed later.
thanks. @normanr @xiaoxiang781216

Reply:
fixed in: https://github.com/apache/incubator-nuttx-apps/pull/958